src/plugins/intel_gpu/src/plugin/graph.cpp

```diff
@@ -133,6 +133,23 @@ std::shared_ptr<cldnn::network> Graph::BuildNetwork(std::shared_ptr<cldnn::progr
     return network;
 }
 
+Graph::memory_states_map Graph::AllocateMemoryStates() {
+    Graph::memory_states_map states {};
+    const auto& memStatesInfo = m_program->GetMemoryStatesInfo();
+    for (const auto& memStateInfo : memStatesInfo) {
+        std::vector<cldnn::layout> orderedLayouts {memStateInfo.second.begin(), memStateInfo.second.end()};
+        std::sort(orderedLayouts.begin(), orderedLayouts.end(), [](cldnn::layout& first, cldnn::layout& second) {
+            return first.size.batch[0] < second.size.batch[0];
+        });
+        std::vector<cldnn::network::MemoryState::Ptr> memoryStates;
+        memoryStates.reserve(orderedLayouts.size());
+        for (const auto& layout : orderedLayouts)
+            memoryStates.push_back(std::make_shared<cldnn::network::MemoryState>(GetEngine()->allocate_memory(layout)));
+        states.insert({memStateInfo.first, memoryStates });
```

Comment:
It would be good to check whether there is already allocated variable with same name

Reply:
`states` is a local variable declared at the beginning of the method body. It's empty when the method starts and gets populated from `memStatesInfo` which is std::map, therefore it cannot contain duplicated names.

Reply:
I got it, let me move the question to the AddMemoryStateInfo.